TMP-LOGGING.md

```diff
@@ -0,0 +1,83 @@
+Logging Guidelines
+==================
+
+## Logging errors
+
+Errors should *always* be logged with `log.Error`, which allows logr
+implementations to provide special handling of errors (for instance,
+providing stack traces in debug mode).
+
+It's acceptible to log call `log.Error` with a nil error object.  This
+conveys that an error occurred in some capacity, but that no actual
+`error` object was involved.
+
+## Logging messages
+
+- Don't put variable content in your messages -- use key-value pairs for
+  that. Never use `fmt.Sprintf` in your message.
+
+- Try to match the terminology in your messages with your tags -- for
+  instance, if you have a tag `api version`, use the term `APIVersion`
+  instead of `GroupVersion` in your message.
+
+## Logging Kubernetes Objects
+
+Kubernetes objects should be logged directly, like `log.Info("this is
+a Kubernetes object", "pod", somePod)`.  controller-runtime provides
+a special encoder for Zap that will transform Kubernetes objects into
+`name, namespace, apiVersion, kind` objects, when available and not in
+development mode.  Other logr implementations should implement similar
+logic.
+
+## Logging Tags
+
```

Comment:
It will be useful to highlight what `Tags` are and how they impact the log message and what is their utility ?

Reply:
replaced the word "tag" with "key-value pairs" to make things more obvious.

Reply:
and expanded on things a bit.

Reply:
Looks good.